src/core/client/ui/types.ts

```diff
@@ -24,3 +24,14 @@ export type Overwrite<T, U> = Pick<T, Exclude<keyof T, keyof U>> & U;
 export type PropTypesOf<T> = T extends React.ComponentType<infer R>
   ? R
   : T extends React.Component<infer S> ? S : {};
+
+/**
+ * Like Partial but applies it deeply.
+ */
+export type DeepPartial<T> = {
```

Comment:
Why duplicate this from https://github.com/coralproject/talk/blob/next/src/core/common/types.ts#L23-L32 ?

Reply:
👍